SimPEG/DataMisfit.py

```diff
@@ -121,11 +121,11 @@ def evalDeriv(self, m, u=None):
         prob   = self.prob
         survey = self.survey
         if u is None: u = prob.fields(m)
-        return prob.Jtvec(m, self.Wd * (self.Wd * survey.residual(m, u=u)), u=u)
+        return prob.Jtvec(m, self.Wd * (self.Wd * survey.residual(m, u)), u)
```

Comment:
this is the topic of #271, and can be addressed in a separate pr so it is consistent across the code base